### PR TITLE
Rahimi-01-urlshortner

### PR DESCRIPTION
**Pull request by SepidehRahimiii**
mvp of urlshortener project

### CODE REVIEW

File: Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/models.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from django.db import models
+
+
+class UrlData(models.Model):    
+    url = models.CharField(max_length=200)
+    slug = models.CharField(max_length=15)
+
+def __str__(self):
```
Comment (ghzamani) [inline Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/models.py:8]: it should be indented

File: Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+def index(request):
+    return HttpResponse("Url Shortener")
+
+def urlShort(request):
+    if request.method == 'POST':
+        form = Url(request.POST)
+        if form.is_valid():
+
+            slug = ''.join(random.choice(string.ascii_letters) for x in range(10))
```
Comment (ghzamani) [inline Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py:20]: how do you check whether the randomly generated slug is unique in database?

**Review by Amirjahandar — COMMENTED** [review]
please mkdir  templates for your form.html and add it in settings.py

File: Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/urls.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+from .models import UrlData
+
+app_name = "url"
+urlpatterns = [
+    path("", views.index, name="home"),
+    path("u/<str:slugs>", views.urlRedirect, name="redirect"),
+    path("", views.urlShort, name = "Short"),
+    path("", UrlData)
+
+]
```
Comment (Amirjahandar) [inline Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/urls.py:12]: you have 3 view for url " "

File: Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    return render(request, 'index.html', context)
+
+def urlRedirect(request, slugs):
+    data = UrlData.objects.get(slug=slugs)
+    return redirect(data.url)
+
+def home_view(request):
+    context ={}
+    context['form']= Url()
+    return render(request, context)
```
Comment (Amirjahandar) [inline Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py:43]: home_view does not have url in urls.py  and  you should return with your template file for exp:
def home_view(request):
    form = Url()
    return render(request, 'form.html', {'form':form}

**Review by Amirjahandar — COMMENTED** [review]
please create .gitignore and add the files that you dont want to track with git

File: Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py (hunk 4 above)
Comment (SepidehRahimiii) [reply Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py:43]: thank rof your commentو
Yes, it 's not complete and I will make corrections soon.

Comment (SepidehRahimiii) [reply Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py:43]: thanks for your comment.
Yes, it's not complete and I will make corrections soon.

File: Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py (hunk 2 above)
Comment (SepidehRahimiii) [reply Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py:20]: I should run a query on database. 
I will create query soon.

Comment (SepidehRahimiii) [reply Zarebin/nextgen-py-5/branches/url/nextgen-py-5/url/views.py:20]: thanks for review, 
I will create a query and check if the slug is unique or not.

**Pull request closed without merging**